### PR TITLE
Runnable code samples!

### PR DESCRIPTION
**Pull request by asyncanup**
and some minor language changes.

Some notes:
- uses RunKit to make code samples runnable
- samples run Node.js based falcor on RunKit Node servers
- also uses a simple now.sh service (https://falcor-server-sample-nvutywuvbl.now.sh/_src) to make `HttpDataSource` samples work

### CODE REVIEW

File: documentation/model.md [hunk 1]
```diff
@@ -1438,32 +1479,37 @@ var model = new falcor.Model({cache: {
...
-            console.error(JSON.stringify(pathValue));
-        });
+var response;
+try {
+    response = await model.
+        treatErrorsAsValues().
+        get('titlesById[44].name');
+} catch (error) {
+    console.log('error was thrown');
+    error
```
Comment (jhusain) [inline documentation/model.md]: Can we make sure to add return statements if the intention is to add data to the console? At the moment this looks like a noop or a bug?

File: documentation/model.md [hunk 2]
```diff
@@ -1438,32 +1479,37 @@ var model = new falcor.Model({cache: {
...
+} finally {
+    console.log('error treated like a value:');
+    response;
```
Comment (jhusain) [inline documentation/model.md]: Please add return statement if possible

File: documentation/model.md (hunk 1 above)
Comment (asyncanup) [reply documentation/model.md]: done

File: documentation/model.md (hunk 2 above)
Comment (asyncanup) [reply documentation/model.md]: done

**Pull request merged**